libs/ledger-live-common/src/modularDrawer/data/entities/interestRateSelectors.ts

```diff
@@ -0,0 +1,42 @@
+import { createSelector } from "@reduxjs/toolkit";
+import { InterestRate } from "../entities";
+
+interface ApiState {
+  assetsDataApi?: {
+    queries?: Record<string, QueryCacheEntry>;
+  };
+}
+
+interface QueryCacheEntry {
+  data?: {
+    pages?: Array<{
+      interestRates?: Record<string, InterestRate>;
+    }>;
+  };
+}
+
+export const selectInterestRateByCurrency: (
```

Comment:
I would have done something like 
```
export const selectInterestRateByCurrency = createSelector(
  [
    (state: ApiState) => state.assetsDataApi?.queries ?? {},
    (_state: ApiState, currencyId: string) => currencyId,
  ],
  (queries, currencyId): InterestRate | undefined => {
  
    for (const query of Object.values(queries)) {
      const pages = query.data?.pages;
      if (!pages) continue;

      for (const page of pages) {
        const rate = page.interestRates?.[currencyId];
        if (rate) return rate;
      }
    }

    return undefined;
  }
);
```

you reduce the number of times you access the value (with intermediate vars) not a big diff but more easy to read no?

Reply:
It's definitely better yes :)